src/xunit.performance.execution/AllocatedBytesForCurrentThread.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Reflection;
+
+namespace Microsoft.Xunit.Performance.Execution
+{
+    /// <summary>
+    ///
+    /// </summary>
+    internal static class AllocatedBytesForCurrentThread
+    {
+        /// <summary>
+        /// Loads the GetAllocatedBytesForCurrentThread method.
+        /// </summary>
+        /// <remarks>
+        /// Calling `gcGetAllocMethod.Invoke(null, null);` will cause an object
+        /// to be returned (it allocates a minimum object), so calling the
+        /// method back to back and computing the difference between them will
+        /// not be equals to zero (it will differ by 24 bytes).
+        /// </remarks>
+        static AllocatedBytesForCurrentThread()
+        {
+            var method = typeof(GC)
+                .GetTypeInfo()
+                .GetMethod("GetAllocatedBytesForCurrentThread", BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic);
+
+            if (method == null)
+            {
+                _GetAllocatedBytesForCurrentThread = () => -0xBAAAAAAD;
+                IsAvailable = false;
+                NoAvailabilityReason = "The running implementation netcoreapp does not expose 'GC.GetAllocatedBytesForCurrentThread'.";
+            }
+            else
+            {
+                _GetAllocatedBytesForCurrentThread = () => (long)method.Invoke(null, null);
+                if (IsBugFixed())
+                {
+                    IsAvailable = true;
+                    NoAvailabilityReason = "";
+                }
+                else
+                {
+                    IsAvailable = false;
```

Comment:
I think for the second case where you encounter the bug - you can point at the issue on GH (show an actual URL) and just say "Not capturing <metric name> because the targeted runtime does not contain the fix for http://github.com/path/to/issue".  Then, it's super clear to users who care about this what to do.